File: backend/src/main/java/com/back/domain/book/book/entity/Book.java
package com.back.domain.book.book.entity;

import com.back.domain.book.category.entity.Category;
import com.back.domain.book.wrote.entity.Wrote;
import com.back.domain.bookmarks.entity.Bookmark;
import com.back.domain.review.review.entity.Review;
import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Book extends BaseEntity {
    @Column(nullable = false)
    String title;
    String imageUrl;
    String publisher;
    int totalPage;
    LocalDateTime publishedDate;
    float avgRate = 0.0f;

    @Column(unique = true)
    String isbn13;

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Wrote> authors = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @OneToMany(mappedBy = "book", orphanRemoval = true)
    private List<Review> reviews = new ArrayList<>();

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Bookmark> bookmarks = new ArrayList<>();

    public Book(String title, String publisher, Category category){
        this.title = title;
        this.publisher = publisher;
        this.category = category;
    }

}

File: backend/src/main/java/com/back/domain/book/book/entity/Book.kt
package com.back.domain.book.book.entity

import com.back.domain.book.category.entity.Category
import com.back.domain.book.wrote.entity.Wrote
import com.back.domain.bookmarks.entity.Bookmark
import com.back.domain.review.review.entity.Review
import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
class Book @JvmOverloads constructor(
    @Column(nullable = false)
    var title: String = "",

    var publisher: String? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = true)
    var category: Category? = null
) : BaseEntity() {

    var imageUrl: String? = null
    var totalPage: Int = 0
    var publishedDate: LocalDateTime? = null
    var avgRate: Float = 0.0f

    @Column(unique = true)
    var isbn13: String? = null

    @OneToMany(mappedBy = "book", cascade = [CascadeType.ALL], orphanRemoval = true)
    private val _authors: MutableList<Wrote> = mutableListOf()

    @OneToMany(mappedBy = "book", orphanRemoval = true)
    private val _reviews: MutableList<Review> = mutableListOf()

    @OneToMany(mappedBy = "book", cascade = [CascadeType.ALL], orphanRemoval = true)
    private val _bookmarks: MutableList<Bookmark> = mutableListOf()

    // Read-only access to collections
    val authors: List<Wrote>
        get() = _authors.toList()

    val reviews: List<Review>
        get() = _reviews.toList()

    val bookmarks: List<Bookmark>
        get() = _bookmarks.toList()

    // Collection management methods
    fun addAuthor(wrote: Wrote) {
        _authors.add(wrote)
    }

    fun removeAuthor(wrote: Wrote) {
        _authors.remove(wrote)
    }

    fun addReview(review: Review) {
        _reviews.add(review)
    }

    fun removeReview(review: Review) {
        _reviews.remove(review)
    }

    fun addBookmark(bookmark: Bookmark) {
        _bookmarks.add(bookmark)
    }

    fun removeBookmark(bookmark: Bookmark) {
        _bookmarks.remove(bookmark)
    }
}